ChildProject/pipelines/samplers.py

```diff
@@ -711,74 +713,65 @@ def _segment_scores(self, recording):
         ).reset_index()
         windows["recording_filename"] = recording["recording_filename"]
 
-        if self.metric == "ctc":
+        if self.metric == "turns":
             if "lena_conv_turn_type" in segments.columns:
                 # NOTE: This is the equivalent of CTC (tab1) in rlena_extract.R
                 segments["is_CT"] = segments["lena_conv_turn_type"].isin(
                     ["TIFR", "TIMR"]
                 )
             else:
                 segments.sort_values(["segment_onset", "segment_offset"])
-                segments = segments[
-                    segments["speaker_type"].isin(["FEM", "MAL", "CHI"])
-                ]
-                segments["duration"] = (
-                    segments["segment_offset"] - segments["segment_onset"]
-                )
+                segments = segments[segments["speaker_type"].isin(self.speakers)]
                 segments["iti"] = segments["segment_onset"] - segments[
                     "segment_offset"
                 ].shift(1)
                 segments["prev_speaker_type"] = segments["speaker_type"].shift(1)
 
-                key_child_env = ["FEM", "MAL", "CHI"]
-                segments["is_CT"] = segments.apply(
-                    lambda row: (row["iti"] < 1000)
-                    and (
-                        (
-                            row["speaker_type"] == "CHI"
-                            and row["prev_speaker_type"] in key_child_env
-                        )
-                        or (
-                            row["speaker_type"] in key_child_env
-                            and row["prev_speaker_type"] == "CHI"
-                        )
-                    ),
-                    axis=1,
+                key_child_environment = set(self.speakers) - {"CHI"}
+
+                segments["is_CT"] = (segments["iti"] < 1000) & (
+                    (
+                        (segments["speaker_type"] == "CHI")
+                        & (segments["prev_speaker_type"].isin(key_child_environment))
+                    )
+                    | (
+                        (segments["speaker_type"].isin(key_child_environment))
+                        & (segments["prev_speaker_type"] == "CHI")
+                    )
                 )
```

Comment:
simplification of lambda/axis - no negative/different behavior consequences that you can think of, right?

Reply:
(oops, that was intended as a comment, not a review item)

Reply:
nope, if anything it should be faster!